### PR TITLE
electron*: refactor 4+ to common expression, add 7 and 8, updates

### LINKED ISSUE #76267: Update all electron versions
**Describe the bug**
^ what is says. They're several months out of date.

https://electronjs.org/releases/stable

Here are the currently supported versions:
https://electronjs.org/docs/tutorial/support#supported-versions

cc @travisbhartwell @manveru

### PR DESCRIPTION
**Pull request by dtzWill**
###### Motivation for this change
Fixes https://github.com/NixOS/nixpkgs/issues/76267

Wanted to add electron_7, duplication and updating was a pain,
so put together a rewrite to make maintaining these easier.

###### Notes
* print-hashes.nix uses upstream-provided SHA256SUMS file
* electron_4: 4.2.8 -> 4.2.12
* electron_5: 5.0.8 -> 5.0.13
* electron_6: 6.0.1 -> 6.1.7
* electron_7:: init at 7.1.10

FWIW:
electron_4 previously did not include at-spi2-core, only at-spi2-atk.
Both are now included, matching other versions for a slightly simpler
expression.

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/development/tools/electron/print-hashes.sh [hunk 1]
```diff
@@ -19,11 +19,15 @@ SYSTEMS=(
     [x86_64-darwin]=darwin-x64
 )
 
+hashfile="$(nix-prefetch-url --print-path "https://github.com/electron/electron/releases/download/v${VERSION}/SHASUMS256.txt" 2>/dev/null|tail -n1)"
```
Comment (worldofpeace) [inline pkgs/development/tools/electron/print-hashes.sh]:
```suggestion
hashfile="$(nix-prefetch-url --print-path "https://github.com/electron/electron/releases/download/v${VERSION}/SHASUMS256.txt" 2>/dev/null| tail -n1)"
```

**Comment by worldofpeace** [comment]
@dtzWill Can you please remove unsupported versions?
* https://www.electronjs.org/docs/tutorial/support#supported-versions

**Comment by dtzWill** [comment]
Maybe a follow-up PR?

I'm not sure I'm up to quality-control/testing everything using electron (which currently defaults to 4, unsupported)-- especially since apparently moving away from 3 broke things folks care about-- hence packaging it too (and likely why we have multiple versions packaged anyway?).

Also, added electron_8! :grin:

**Comment by worldofpeace** [comment]
> Maybe a follow-up PR?
> 
> I'm not sure I'm up to quality-control/testing everything using electron (which currently defaults to 4, unsupported)-- especially since apparently moving away from 3 broke things folks care about-- hence packaging it too (and likely why we have multiple versions packaged anyway?).
> 
> Also, added electron_8! grin

Ugh, I did forget about this. I've handled the prior electron updates, it's a bit of a pain having to test the consumers per upgrades/seeing what electron versions they should be users. Yep, we have that as a followup.

**Review by worldofpeace — APPROVED** [review]

**Comment by worldofpeace** [comment]
Thanks @dtzWill, this is mine and your's todo lists now :smile:

**Pull request merged**